sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs

```diff
@@ -24,8 +27,16 @@ public DiagnosticsOptions()
         /// </summary>
         public bool IsLoggingContentEnabled { get; set; }
 
+        /// <summary>
+        /// Gets or sets value indicating maximum size of content to log in bytes. Defaults to 4096.
+        /// </summary>
         public int LoggingContentSizeLimit { get; set; } = 4 * 1024;
 
+        public IList<string> LoggingAllowedHeaders { get; }
```

Comment:
Nit - should this be `HeaderNames`?  I don't feel strongly, but everything else called `Headers` is `IEnumerable<HttpHeader>`